### DISCUSSION TITLE
What is the name of the com.amazonaws.auth.policy in AWS SDK Java 2.x?

### DISCUSSION THREAD

**Discussion by yyjyijie** [aws/aws-sdk-java-v2#3833] [Migration from v1 to v2]
I am migrating from aws sdk java v1 to v2, does anyone know what is the following module names in v2:

```
import com.amazonaws.auth.policy.Action;
import com.amazonaws.auth.policy.Policy;
import com.amazonaws.auth.policy.Statement;
import com.amazonaws.auth.policy.actions.S3Actions;
```
And is AWSSessionCredentialsProvider supported in AWS SDK Java v2?

**Comment by debora-ito** [comment]
Those packages were part of the IAM Policy Builder, and it's not yet supported in v2.

We are tracking the feature request here: https://github.com/aws/aws-sdk-java-v2/issues/39

**Reply by yyjyijie** [reply]
Is there any estimation when it will be supported in v2?

And in this sense, in order to use the IAM policy, should I still use aws sdk java v1?

And how about AWSSessionCredentialsProvider? Do we support it in v2?

**Reply by debora-ito** [reply]
> Is there any estimation when it will be supported in v2?

No, usually we don't comment about timelines, and we don't have a public backlog, unfortunately.

> And in this sense, in order to use the IAM policy, should I still use aws sdk java v1?

Yes, and you can use Java v1 and v2 side-by-side.

> And how about AWSSessionCredentialsProvider? Do we support it in v2?

We don't have an equivalent for the AWSSessionCredentialsProvider interface, I recommend you take a look at the list of available credential providers in [AwsCredentialsProvider](http://docs.aws.amazon.com/sdk-for-java/latest/reference/software/amazon/awssdk/auth/credentials/AwsCredentialsProvider.html), see All Known Implementing Classes.
